lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -84,7 +126,30 @@ gen_stats! {
     shader_compile_count : usize,
 }
 
+
+// === StatsData methods ===
+
 impl StatsData {
+    fn begin_frame(&mut self, time: f64) {
+        self.frame_counter += 1;
+
+        if self.frame_begin_time > 0.0 {
+            let end_time = time;
```

Comment:
this if block should have a short doc describing what it does. I know it was not in the previous codebase, but it should have one.

Reply:
It's perfect